### ISSUE TITLE
Terraform refresh/plan/apply hangs on when using uuid()

### ISSUE THREAD

**Issue by gurdulu** [hashicorp/terraform#11635]
Hi there,

I have tried to use uuid() for generating a unique filename, what follows is a simplified example but it reproduces perfectly the issue
```hcl
variable "filename" {}
resource "null_resource" "zip" {
  provisioner "local-exec" {
    command = "zip -r ${var.filename}.zip main.tf"
  }
}
```
Invoking the module 
```hcl
module "zip1" {
  source = "modules/example"
  filename = "test-${uuid()}"
}
```
launching terraform with plan/apply, it hangs on indefinitely.

Terraform Version: 0.8.4 on El Capitan 10.11.6 and 0.8.5 on RHEL7

**Comment by grubernaut** [comment]
Hi @gurdulu, thanks for the issue! 

It doesn't solve the issue, but I was able to workaround it by using the `timestamp()` interpolation function inside the `local-exec` provisioner.

**Comment by gurdulu** [comment]
Hi @grubernaut 

Thank you for the tip. 

I used a slightly different approach for being sure the value is different each time the module is invoked

```hcl
resource "random_id" "random" {
  keepers {
    tm = "${timestamp()}"
  }
  byte_length = 8
}
```

and then using the hexadecimal output of the resource

```hcl
filename = "file-${random_id.random2.hex}"
```

**Comment by mitchellh** [comment]
Will be fixed by #11733. Thanks!
